src/app/clusters/resource-monitoring-server/ResourceMonitoringCluster.cpp

```diff
@@ -16,94 +14,184 @@
 
 using namespace chip::app;
 using namespace chip::app::Clusters;
 using namespace chip::app::Clusters::ResourceMonitoring;
+using namespace chip::app::Clusters::ResourceMonitoring::Attributes;
 using chip::Protocols::InteractionModel::Status;
 
 namespace chip {
 namespace app {
 namespace Clusters {
 namespace ResourceMonitoring {
 
-Instance::Instance(Delegate * aDelegate, EndpointId aEndpointId, ClusterId aClusterId, uint32_t aFeatureMap,
-                   ResourceMonitoring::Attributes::DegradationDirection::TypeInfo::Type aDegradationDirection,
-                   bool aResetConditionCommandSupported) :
-    CommandHandlerInterface(Optional<EndpointId>(aEndpointId), aClusterId),
-    AttributeAccessInterface(Optional<EndpointId>(aEndpointId), aClusterId), mDelegate(aDelegate), mEndpointId(aEndpointId),
-    mClusterId(aClusterId), mDegradationDirection(aDegradationDirection), mFeatureMap(aFeatureMap),
-    mResetConditionCommandSupported(aResetConditionCommandSupported)
+ResourceMonitoringCluster::ResourceMonitoringCluster(
+    EndpointId aEndpointId, ClusterId aClusterId, const BitFlags<ResourceMonitoring::Feature> enabledFeatures,
+    OptionalAttributeSet optionalAttributeSet,
+    ResourceMonitoring::Attributes::DegradationDirection::TypeInfo::Type aDegradationDirection,
+    bool aResetConditionCommandSupported) :
+    DefaultServerCluster(ConcreteClusterPath(aEndpointId, aClusterId)),
+    mDegradationDirection(aDegradationDirection), mResetConditionCommandSupported(aResetConditionCommandSupported),
+    mEnabledFeatures(enabledFeatures), mOptionalAttributeSet(optionalAttributeSet)
+{}
+
+CHIP_ERROR ResourceMonitoringCluster::SetDelegate(Delegate * aDelegate)
 {
+    VerifyOrReturnError(aDelegate != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+
+    mDelegate = aDelegate;
     mDelegate->SetInstance(this);
+    return mDelegate->Init();
+}
+
+DataModel::ActionReturnStatus ResourceMonitoringCluster::WriteAttribute(const DataModel::WriteAttributeRequest & request,
+                                                                        AttributeValueDecoder & decoder)
+{
+    return NotifyAttributeChangedIfSuccess(request.path.mAttributeId, WriteImpl(request, decoder));
 }
 
-Instance::~Instance()
+DataModel::ActionReturnStatus ResourceMonitoringCluster::WriteImpl(const DataModel::WriteAttributeRequest & request,
+                                                                   AttributeValueDecoder & decoder)
 {
-    if (mDelegate)
+    AttributePersistence persistence{ mContext->attributeStorage };
+
+    switch (request.path.mAttributeId)
     {
-        mDelegate->SetInstance(nullptr);
+    case ResourceMonitoring::Attributes::LastChangedTime::Id: {
+        ReturnErrorOnFailure(chip::app::GetSafeAttributePersistenceProvider()->WriteScalarValue(
+            ConcreteAttributePath(GetEndpointId(), GetClusterId(), Attributes::LastChangedTime::Id), mLastChangedTime));
+        return Protocols::InteractionModel::Status::Success;
```

Comment:
```suggestion
        return chip::app::GetSafeAttributePersistenceProvider()->WriteScalarValue(
            ConcreteAttributePath(GetEndpointId(), GetClusterId(), Attributes::LastChangedTime::Id), mLastChangedTime);
```

seems more concise. CHIP_NO_ERROR and Status::Success are generally equivalent